src/bridge/MatrixHandler.ts

```diff
@@ -23,6 +23,25 @@ async function reqHandler(req: BridgeRequest, promise: PromiseLike<unknown>) {
     }
 }
 
+/**
+ * Returns a trimmed string if the input text is all Markdown-style code.
+ * This is used to allow small code snippets to look nice in IRC and not be
+ * transformed into an upload.
+ */
+function trimCodeMessages(text: string): string|undefined {
+    let trimmedText = text.trim();
+    // If this post isn't all code, ignore it.
+    if (!/^```.*\n.*```$/s.test(trimmedText)) {
+        return undefined;
+    }
+    // Remove the first line (e.g. ```js) and the ``` at the end
+    trimmedText = trimmedText.substring(trimmedText.indexOf('\n'), trimmedText.length - 3).trim();
+    // Add monospace character at the start of each line
+    // https://modern.ircdocs.horse/formatting.html#monospace
+    trimmedText = trimmedText.replace(/\n/, '\n\x11');
```

Comment:
oooh fancy

Reply:
I don't know how to test this, so I removed it for now.